rapido.css

```diff
@@ -225,39 +225,51 @@ article.rapido > footer > section {
 /* div */
 
 .rapido div {
-	width: 100%;
-	max-width: 600px;
+	display: block;
+	flex: 0 0 auto;
+	max-width: 100%;
+	margin: 0 20px 0 0;
 	overflow-y: hidden;
 	overflow-x: auto;
 }
 
-.rapido section figure > div {
+.rapido > section div {
 	width: 600px;
-	max-width: 100%;
-	overflow-y: hidden;
-	overflow-x: auto;
 }
 
-/* end div */
+.rapido > section > div {
+	margin: 0 0 20px 0;
+}
 
-/* figcaption */
+.rapido > footer div,
+article.rapido > header div {
+	width: 100%;
+}
 
-.rapido section figcaption {
-	width: 324px;
-	margin-left: 20px;
+article.rapido > header > div {
+	margin: 0 0 20px 0;
 }
 
-@media (max-width: 1023px) {
-	.rapido section figcaption {
-		width: 600px;
-		margin: 5px 0 0 0;
+.rapido > footer > section > div {
+	margin: 0 0 10px 0;
+}
+
+@media (min-width: 1024px) {
+	.rapido > footer > section > div {
+		width: calc(100% - 200px);
+		margin: 0 0 10px 200px;
 	}
 }
 
-article.rapido > header > figure > figcaption,
-article.rapido > footer > section > figure > figcaption {
-	width: 100%;
-	margin: 5px 0 0 0;
+/* end div */
+
+/* figcaption */
+
+.rapido figcaption {
+	flex: 1 1;
+	min-width: 140px;
+	max-width: 100%;
+	margin-top: 3px;
```

Comment:
Captions grow or shrink according to the image width, for a minimum width of 140px.